src/hypothesis/strategies.py

```diff
@@ -1279,6 +1389,42 @@ def __not_a_first_class_strategy(self, name):
             ) % (name,))
     return DataStrategy()
 
+
+def register_type_strategy(custom_type, strategy):
+    """Add an entry to the global type-to-strategy lookup.
+
+    This lookup is used in :func:`~hypothesis.strategies.builds` and
+    :func:`@given <hypothesis.given>`.
+
+    :func:`~hypothesis.strategies.builds` will be used automatically for
+    classes with type annotations on ``__init__`` , so you only need to
+    register a strategy if one or more arguments need to be more tightly
+    defined than their type-based default, or if you want to supply a strategy
+    for an argument with a default value.
+
+    ``strategy`` may be:
+
+    - a search strategy
+    - a function that takes a type and returns a strategy (useful for generic
+      types where unknown subtypes can be handled automatically)
+    - ``None``, in which case ``custom_type`` is removed from the lookup
```

Comment:
This feels like a really weird API to me. I'd rather have separate register/unregister functions I think